analyzers/src/SonarAnalyzer.Common/Rules/CommentsShouldNotBeEmptyBase.cs

```diff
@@ -0,0 +1,126 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class CommentsShouldNotBeEmptyBase<TSyntaxKind> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+{
+    private const string DiagnosticId = "S4663";
+
+    protected abstract string GetCommentText(SyntaxTrivia trivia);
+    protected abstract bool IsValidTriviaType(SyntaxTrivia trivia);
+
+    protected abstract bool IsSimpleComment(SyntaxTrivia trivia);
+    protected abstract bool IsEndOfLine(SyntaxTrivia trivia);
+    protected abstract bool IsWhitespace(SyntaxTrivia trivia);
+
+    protected override string MessageFormat => "Remove this empty comment";
+
+    protected CommentsShouldNotBeEmptyBase() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterTreeAction(Language.GeneratedCodeRecognizer, c =>
+        {
+            foreach (var token in c.Tree.GetRoot().DescendantTokens())
+            {
+                CheckTrivia(c, token.LeadingTrivia);
+                CheckTrivia(c, token.TrailingTrivia);
+            }
+        });
+
+    protected void CheckTrivia(SonarSyntaxTreeReportingContext context, IEnumerable<SyntaxTrivia> trivia)
+    {
+        foreach (var partition in Partition(trivia).Where(ShouldReport))
+        {
+            var start = partition.First().GetLocation().SourceSpan.Start;
+            var end = partition.Last().GetLocation().SourceSpan.End;
+
+            var location = Location.Create(context.Tree, TextSpan.FromBounds(start, end));
+            context.ReportIssue(Diagnostic.Create(Rule, location));
+        }
+    }
+
+    protected IEnumerable<IEnumerable<SyntaxTrivia>> Partition(IEnumerable<SyntaxTrivia> trivia)
+    {
+        var res = new List<List<SyntaxTrivia>>();
+
+        var current = new List<SyntaxTrivia>();
+        var endOfLineFound = false;
+
+        foreach (var trivium in trivia)
+        {
+            if (IsWhitespace(trivium))
+            {
+                continue;
+            }
+
+            if (IsSimpleComment(trivium)) // put it on the current block of "//"
+            {
+                current.Add(trivium);
+                endOfLineFound = false;
+                continue;
+            }
+
+            // This is for the case, of two different comment types, for example:
+            // //
+            // ///
+            if (IsValidTriviaType(trivium)) // valid but not "//", because of the upper if
+            {
+                AddCurrent();
+                res.Add(new List<SyntaxTrivia> { trivium });
+            }
+            else if (IsEndOfLine(trivium))
+            {
+                // This is for the case, of an empty line in between, for example:
+                // //
+                //
```

Comment:
```suggestion
                // //  \n <- here endOfLineFound is false and set to true
                // \n <- here endOfLineFound is true
```

Reply:
I'm not sure I understand this case.
Is it to check that the literal "\n" is not confused as an EOL?

Should i add it WITHOUT the `<- here...` part?

Reply:
It's meant to be an explanation about the state tracked with `endOfLineFound`. The `\n` is there to make it more obvious what we are looking at right now (It's hard to illustrate whitespace). If it is more confusing than helpful rework it.